pkg/cloudprovider/cloudprovider.go

```diff
@@ -109,7 +110,7 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 		return nil, cloudprovider.NewNodeClassNotReadyError(stderrors.New(nodeClassReady.Message))
 	}
 	if nodeClassReady.IsUnknown() {
-		return nil, fmt.Errorf("resolving NodeClass readiness, NodeClass is in Ready=Unknown, %s", nodeClassReady.Message)
+		return nil, cloudprovider.NewCreateError(fmt.Errorf("resolving NodeClass readiness, NodeClass is in Ready=Unknown, %s", nodeClassReady.Message), consts.NodeClassReadinessUnknownReason, nodeClassReady.Message)
```

Comment:
`consts.NodeClassReadinessUnknownReason` (and other new consts) are only used here. I don't see a good reason to tuck them away in const package. Would suggest moving them in here (or even getting rid of the const altogether).

Reply:
Moved all of the consts to the cloudprovider.go. (Didn't delete it for now, since there were other unrelated const variables) 

Later when we get too much error reasons, we could move them to a separate error consts file just to keep track of them, but that's in the far future from what I can see.